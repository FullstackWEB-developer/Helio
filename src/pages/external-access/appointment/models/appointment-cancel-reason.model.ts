export interface AppointmentCancelReason {
    appointmentCancelReasonId: number;
    name: string;
    noShow: boolean;
    patientRescheduled: boolean;
    providerUnavailable: boolean;
    slotAvailable: boolean;
}
