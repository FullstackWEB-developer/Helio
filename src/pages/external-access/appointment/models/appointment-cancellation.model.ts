export interface AppointmentCancellationModel {
    patientId: number;
    appointmentCancelReasonId: number;
    cancellationReason: string;
}
