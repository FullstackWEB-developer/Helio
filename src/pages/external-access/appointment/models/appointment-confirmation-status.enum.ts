export enum ConfirmationStatus {
    ConfirmedViaSMS = 161,
    ConfirmedViaEmail = 121
}