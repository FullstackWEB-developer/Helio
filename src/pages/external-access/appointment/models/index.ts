export * from './appointment-type.model';
export * from './appointment-slot.model';
export * from './appointment.model';
export * from './appointment-cancellation.model';
export * from './appointment-cancel-reason.model';
export * from './appointment-slot-request.model';
