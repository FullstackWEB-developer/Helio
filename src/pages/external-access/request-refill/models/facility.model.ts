export interface Facility {
    ncpdpId: number;
    facilityId: number;
    name: string;
    city: string;
    pharmacyType: string;
    faxNumber: string;
    address: string;
    distance: string;
    zipCode: string;
    state: string;
    phoneNumber: string;
}
