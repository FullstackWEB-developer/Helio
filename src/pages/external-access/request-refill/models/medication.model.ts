export interface Medication {
    medicationId: number;
    medicationName: string;
    refillsAllowed: number;
}
