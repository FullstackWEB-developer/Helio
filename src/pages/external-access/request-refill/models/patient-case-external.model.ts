export enum PatientCaseDocumentSubClass {
    Refill = 1,
    PatientCase
}

export enum PatientCaseDocumentSource {
    Portal = 1,
    Partner
}

export interface PatientCaseExternal {
    departmentId: number;
    providerId: number;
    internalNote: string;
    ignoreNotification: boolean;
    documentSubClass: PatientCaseDocumentSubClass;
    documentSource: PatientCaseDocumentSource;
}

export interface PatientCaseCreateProps {
    patientId: number;
    patientCaseExternal: PatientCaseExternal
}
