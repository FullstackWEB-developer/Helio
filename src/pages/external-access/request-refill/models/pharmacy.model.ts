export interface Pharmacy {
    pharmacyType?: string;
    state: string;
    city: string;
    receiverType?: string;
    acceptFax?: boolean;
    clinicalProviderId: number;
    zip: string;
    phoneNumber: string;
    clinicalProviderName: string;
    address1: string;
    address2?: string;
    faxNumber: string;
    suite?: string;
}
