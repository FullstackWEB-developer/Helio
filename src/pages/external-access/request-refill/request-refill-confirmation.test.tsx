import {unmountComponentAtNode} from 'react-dom';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import React from 'react';
import i18n from '../../../i18nForTests';
import MockDate from 'mockdate';
import TestWrapper from '@shared/test-utils/test-wrapper';
import {act, render} from '@testing-library/react';
import {VerifiedPatient} from '@pages/patients/models/verified-patient';
import {Provider} from '@shared/models';
import RequestRefillConfirmation from '@pages/external-access/request-refill/request-refill-confirmation';
describe("Request Refill Confirmation tests", () => {

    let container: HTMLDivElement | null;
    const mockState = {
        patientsState: {
            verifiedPatient: {
                patientId: 3,
                firstName: 'Burak',
                lastName: 'BOLEK',
                departmentId: 3
            } as VerifiedPatient
        },
        lookupsState: {
            providerList: [
                {
                    ansiNameCode: '',
                    ansiSpecialtyCode: '',
                    billable: false,
                    createEncounterOnCheckin: false,
                    firstName: 'provider',
                    lastName: 'lastname',
                    id: 3
                }
            ] as Provider[]
        },
    };
    beforeEach(async () => {
        await i18n.init();
        MockDate.set(dayjs('2018-04-04T16:00:00.000Z').toDate());
        dayjs.extend(duration);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
        MockDate.reset();
    });


    it("renders Request Refill Confirmation correctly", async () => {
        let fragment: any;
        await act(async () => {
            const {asFragment, findByText} = render(<TestWrapper mockState={mockState}>
                <RequestRefillConfirmation/>
            </TestWrapper>)
            fragment = asFragment;
            expect(await findByText('external_access.medication_refill.back_to_medications')).not.toBeNull();
            expect(await findByText('external_access.medication_refill.refill_request_sent')).not.toBeNull();
            expect(await findByText('external_access.medication_refill.refill_request_sent_details')).not.toBeNull();
            await new Promise((r) => setTimeout(r, 2600));
            //Hipaa dialog is displayed after 2.5 seconds
            expect(await findByText('common.close_window_title')).not.toBeNull();
        });
        expect(fragment()).toMatchSnapshot();
    });
});
