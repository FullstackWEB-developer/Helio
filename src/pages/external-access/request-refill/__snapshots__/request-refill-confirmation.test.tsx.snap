// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Refill Confirmation tests renders Request Refill Confirmation correctly 1`] = `
<DocumentFragment>
  <div
    class="2xl:px-48 pt-7 without-default-padding"
  >
    <a
      href="/o/view-medications"
    >
      <div
        class="flex flex-row pb-5 cursor-pointer"
      >
        <div
          class=""
        >
          <svg
            class="icon-medium "
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="24"
              width="24"
            />
            <path
              class="fill-default"
              d="M-1.408,10.085H-19.121l5.32,5.052a1.324,1.324,0,0,1,.419.96,1.324,1.324,0,0,1-.419.96,1.468,1.468,0,0,1-1.01.4,1.468,1.468,0,0,1-1.01-.4l-7.76-7.37a1.311,1.311,0,0,1,0-1.92L-15.822.4A1.463,1.463,0,0,1-14.811,0,1.462,1.462,0,0,1-13.8.4a1.324,1.324,0,0,1,.419.96,1.324,1.324,0,0,1-.419.96l-5.32,5.053H-1.408A1.394,1.394,0,0,1,0,8.727,1.394,1.394,0,0,1-1.408,10.085Z"
              transform="translate(24 3)"
            />
          </svg>
        </div>
        <div
          class="body2 pl-4"
        >
          external_access.medication_refill.back_to_medications
        </div>
      </div>
    </a>
    <div
      class="2xl:whitespace-pre 2xl:h-12 2xl:my-3 flex w-full items-center"
    >
      <h4>
        external_access.medication_refill.refill_request_sent
      </h4>
    </div>
    <div
      class="pt-9 pb-8"
    >
      external_access.medication_refill.refill_request_sent_details
    </div>
    <div
      class="flex items-center justify-center m-auto justify-self-center"
    >
      <div
        class="flex items-center justify-center w-full h-full"
      >
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
